mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -1039,19 +1039,46 @@ impl<T: Send, E: Send> Future for S3MetaRequest<T, E> {
     }
 }
 
-#[pinned_drop]
-impl<T, E> PinnedDrop for S3MetaRequest<T, E> {
-    fn drop(self: Pin<&mut Self>) {
-        self.meta_request.cancel();
-    }
-}
-
 impl<T: Send, E: Send> FusedFuture for S3MetaRequest<T, E> {
     fn is_terminated(&self) -> bool {
         self.receiver.is_terminated()
     }
 }
 
+/// Wrapper for a [MetaRequest] that cancels it on drop.
```

Comment:
Are there any reason why we don't want this behavior to bake into `MetaRequest` itself?

Reply:
`MetaRequest` is essentially a smart pointer for an `aws_s3_meta_request`. It implements `Clone` (`aws_s3_meta_request_acquire`) and `Drop` (`aws_s3_meta_request_release`). There are cases where we internally use multiple clones and don't want them to cancel the request on drop.

Reply:
It may be worth documenting that `MetaRequest` won't be cancelled on drop and the callers must ensure it's dropped.